CLI/EDDBFormatter/__main__.py

```diff
@@ -99,15 +107,26 @@ def run_eddb():
         }
         # To be used as a diversion station, must have L pad, Repair function,
         # not on a planet, not a mobile platform, and no more than 800 LS from the main star.
-        if (
-            temp_station_dict["max_landing"] == "L"
-            and temp_station_dict["has_repair"]
-            and not temp_station_dict["is_planet"]
-            and temp_station_dict["station_type"] != "Fleet Carrier"
-            and temp_station_dict["dist_star"] <= 800
-        ):
-            station_dict[counter] = temp_station_dict
-            counter += 1
+        if filter_dist:
+            if (
+                temp_station_dict["max_landing"] == "L"
+                and temp_station_dict["has_repair"]
+                and not temp_station_dict["is_planet"]
+                and temp_station_dict["station_type"] != "Fleet Carrier"
+                and temp_station_dict["dist_star"] <= 800
+            ):
+                station_dict[counter] = temp_station_dict
+                counter += 1
+        else:
+            if (
+                temp_station_dict["max_landing"] == "L"
+                and temp_station_dict["has_repair"]
+                and not temp_station_dict["is_planet"]
+                and temp_station_dict["station_type"] != "Fleet Carrier"
+            ):
+                station_dict[counter] = temp_station_dict
+                counter += 1
+
```

Comment:
lots of duplicated logic here, consider refactoring.
you can adapt the input iterable using a generator expression prior to for looping over it.
something to the effect of
```py
data = tqdm(data, desc= ... )
data = (item for item in data if item["max_landing" == "L")
data = (item for item in data if ... )
...
for temp_station_dict in data:
```

note the use of `()`, this is a *generator expression*, which basically allows you to wrap one iterable in another iterable.
The above example uses several of these chained together to apply the requisite filtering logic.
Generator expressions do not add time complexity to the algorithm, as opposed to list comprehensions which share very similar syntax.